tests/ci/cdk/pipeline/scripts/build_target.sh

```diff
@@ -0,0 +1,187 @@
+#!/usr/bin/env bash
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0 OR ISC
+
+set -exuo pipefail
+
+source util.sh
+
+echo \"Environment variables:\"
+env
+
+if [[ -z "${NEED_REBUILD+x}" || -z "${NEED_REBUILD}" || ${NEED_REBUILD} -eq 0 ]]; then
+  echo "No rebuild needed"
+  exit 0
+fi
+
+export COMMIT_HASH=${COMMIT_HASH:-$CODEBUILD_RESOLVED_SOURCE_VERSION}
+export CROSS_ACCOUNT_BUILD_ROLE_ARN="arn:aws:iam::${DEPLOY_ACCOUNT}:role/CrossAccountCodeBuildRole"
+export CROSS_ACCOUNT_BUILD_SESSION="pipeline-${COMMIT_HASH}"
+
+function build_codebuild_ci_project() {
+  local attempt=0
+  local project=${1}
+
+  if [[ -z ${project} ]]; then
+    echo "No project name provided."
+    exit 1
+  fi
+
+  if [[ ${DEPLOY_ACCOUNT} == '351119683581' ]]; then
+    source_version="main"
+  else
+    source_version=${COMMIT_HASH}
+  fi
+
+  echo "Starting CI tests in ${project}"
+  start_codebuild_project "${project}" "${source_version}"
+
+  while [[ ${attempt} -le ${MAX_RETRY} ]]; do
+    if [[ $attempt -gt 0 ]]; then
+      echo "Retrying ${attempt}/${MAX_RETRY}..."
+    fi
```

Comment:
Should you move this to be after line 49? For example `attempt == 5` and `max_retry == 5`. You will log "retrying 5/5", then increment so that `attempt == 6` and then you would fail the check on line 49. So you didn't really make another attempt.